src/main/java/com/google/api/generator/engine/ast/TypeNode.java

```diff
@@ -155,4 +187,17 @@ private static TypeNode createPrimitiveArrayType(TypeKind typeKind) {
   private static boolean isPrimitiveType(TypeKind typeKind) {
     return !typeKind.equals(TypeKind.OBJECT);
   }
+
+  private static HashMap<TypeNode, TypeNode> createBoxedTypeMap() {
```

Comment:
We generally return the superinterface when it comes to collections, can we return a Map here instead?

Reply:
noted, done.